clients/python-wrapper/lakefs/branch.py

```diff
@@ -130,9 +130,88 @@ def object(self, path: str) -> WriteableObject:
 
         return WriteableObject(self.repo_id, self._id, path, client=self._client)
 
-    @property
-    def objects(self) -> WriteableObjectManager:
+    def uncommitted(self, max_amount: Optional[int], after: str = '', prefix: str = '') -> Generator[Change]:
         """
-        Return an Objects object using the current repository id and client
+        List uncommitted changes
+
+        :param max_amount: maximum amount of changes to return
+        :param after: return changes after this path
+        :param prefix: return changes with this prefix
+        :return: a generator of Change objects
+        :raises:
+            NotFoundException if branch or repository do not exist
+            NotAuthorizedException if user is not authorized to perform this operation
+            ServerException for any other errors
         """
-        return WriteableObjectManager()
+        for diff in self._get_generator(self._client.sdk_client.branches_api.diff_branch,
+                                        self._repo_id, self._id, max_amount=max_amount, after=after, prefix=prefix):
+            yield Change(**diff.dict())
+
+    def import_data(self, commit_message: str) -> ImportManager:
+        """
+        Import data to lakeFS
+
+        :param commit_message: once the data is imported, a commit is created with this message
+        :return: an ImportManager object
+        """
+        return ImportManager(self._repo_id, self._id, commit_message, self._client)
+
+    def delete_objects(self, object_paths: str | Iterable[str]) -> None:
+        """
+        Delete objects from lakeFS
+
+        :param object_paths: a single path or an iterable of paths to delete
+        :return: None
+        :raises:
+            NotFoundException if branch or repository do not exist
+            NotAuthorizedException if user is not authorized to perform this operation
+            ServerException for any other errors
+        """
+        if isinstance(object_paths, str):
+            object_paths = [object_paths]
+        with api_exception_handler():
+            return self._client.sdk_client.objects_api.delete_objects(
+                self._repo_id,
+                self._id,
+                lakefs_sdk.PathList(paths=object_paths)
+            )
+
+    def transact(self, commit_message: str) -> Transaction:
+        """
+        Create a transaction for multiple operations
+
+        :param commit_message: once the transaction is committed, a commit is created with this message
+        :return: a Transaction object to perform operations on
+        """
+        return Transaction(self._repo_id, self.id, commit_message, self._client)
+
+    def reset_changes(self, path_type: Literal["common_prefix", "object", "reset"] = "reset",
```

Comment:
What about the idea of creating a namedtuple? This can create strange configuration such as path=None and path_type=object.
We should either couple the two variables (via namedtuple or otherwise) or add verification

Reply:
It respects the API, I don't think we should verify this. Also, I didn't like the option with the namedtuple because it's not "typed", and using a class for this also felt like an overkill. Honestly, I think that sticking with the API is the right thing to do here, but if you have strong thoughts about it, I can change it.

Reply:
Not blocking on this